graylog2-server/src/test/java/org/graylog/plugins/views/migrations/V20191203120602_MigrateSavedSearchesToViewsSupport/V20191203120602_MigrateSavedSearchesToViewsTest.java

```diff
@@ -0,0 +1,174 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+public class V20191203120602_MigrateSavedSearchesToViewsTest {
+    @Rule
+    public final MongoDBInstance mongodb = MongoDBInstance.createForClass();
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private ClusterConfigService clusterConfigService;
+    @Mock
+    private SearchService searchService;
+    @Mock
+    private ViewService viewService;
+
+    private Migration migration;
+    private final ObjectMapper objectMapper = new ObjectMapperProvider().get();
+
+    static class StaticRandomObjectIdProvider extends RandomObjectIdProvider {
+        private final Date date;
+        private AtomicInteger counter;
+
+        StaticRandomObjectIdProvider(Date date) {
+            super(date);
+            this.date = date;
+            this.counter = new AtomicInteger(0);
+        }
+
+        @Override
+        public String get() {
+            return new ObjectId(date, 42, (short) 23, counter.incrementAndGet()).toHexString();
+        }
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        final MongoJackObjectMapperProvider mapperProvider = new MongoJackObjectMapperProvider(new ObjectMapper());
+
+        final SavedSearchService savedSearchService = new SavedSearchService(mongodb.mongoConnection(), mapperProvider);
+        final RandomObjectIdProvider randomObjectIdProvider = new StaticRandomObjectIdProvider(new Date(1575020937839L));
+        final RandomUUIDProvider randomUUIDProvider = new RandomUUIDProvider(new Date(1575020937839L), 1575020937839L);
+
+        this.migration = new V20191203120602_MigrateSavedSearchesToViews(
+                clusterConfigService,
+                savedSearchService,
+                searchService,
+                viewService,
+                randomObjectIdProvider,
+                randomUUIDProvider
+        );
+    }
+
+    @Test
+    public void runsIfNoSavedSearchesArePresent() {
+        this.migration.upgrade();
+    }
+
+    @Test
+    public void writesMigrationCompletedAfterSuccess() {
+        this.migration.upgrade();
+
+        final MigrationCompleted migrationCompleted = captureMigrationCompleted();
+        assertThat(migrationCompleted.savedSearchIds()).isEmpty();
+    }
+
+    @Test
```

Comment:
I would prefer writing different tests for the different kinds of saved searches instead of one big test. I had to test all three timerange variants manually too.

Reply:
✅

Reply:
Nice! Is there a special reason for not deduplicating the code by e.g. extracting a method that takes the string identifiers for fixtures and entities as parameters?
One parameterized test might also work, if we have an alternative way to load the fixture than the `MongoDBFixtures` annotation.

Reply:
There is no special reason. Deduplicating the code would make sense if we can expect to change the code afterwards, which I do not for a one-off migration.

Reply:
Ok, fair enough.